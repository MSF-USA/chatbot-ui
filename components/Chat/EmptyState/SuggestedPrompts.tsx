'use client';

import React, { useMemo } from 'react';
import { suggestedPrompts } from '@/components/Chat/prompts';

interface SuggestedPromptsProps {
  onSelectPrompt?: (prompt: string) => void;
  count?: number;
}

/**
 * Display suggested prompts for new conversations
 */
export function SuggestedPrompts({
  onSelectPrompt,
  count = 3,
}: SuggestedPromptsProps) {
  // Use first N prompts to avoid hydration mismatch from randomization
  const displayedPrompts = useMemo(() => {
    return suggestedPrompts.slice(0, count);
  }, [count]);

  return (
    <div className="hidden sm:flex space-x-5">
      {displayedPrompts.map((prompt, index) => {
        const Icon = prompt.icon;

        return (
          <button
            key={index}
            className="bg-transparent text-black dark:text-white border border-[#E0E0E0] dark:border-[#444444] rounded-md px-2 py-1 text-sm hover:bg-[#F9F9F9] dark:hover:bg-[#2F2F2F] dark:hover:text-white transition"
            onClick={() => onSelectPrompt?.(prompt.prompt)}
            style={{
              width: '200px',
              height: '100px',
              textAlign: 'start',
              whiteSpace: 'normal',
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'start',
              justifyContent: 'center',
              padding: '30px',
            }}
          >
            {Icon && (
              <div className="flex flex-col items-start">
                <Icon className="h-5 w-5 mb-2" />
                <div>
                  <span>{prompt.title}</span>
                </div>
              </div>
            )}
          </button>
        );
      })}
    </div>
  );
}
