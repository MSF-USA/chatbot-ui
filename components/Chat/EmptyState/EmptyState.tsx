'use client';

import { useTranslations } from 'next-intl';
import { SuggestedPrompts } from './SuggestedPrompts';
import Image from 'next/image';
import { useUI } from '@/lib/hooks/ui/useUI';
import lightTextLogo from '@/public/international_logo_black.png';
import darkTextLogo from '@/public/international_logo_white.png';

interface EmptyStateProps {
  onSelectPrompt?: (prompt: string) => void;
}

/**
 * Empty state shown when no messages in conversation
 */
export function EmptyState({ onSelectPrompt }: EmptyStateProps) {
  const t = useTranslations();
  const { theme } = useUI();

  return (
    <div className="flex h-[88%] items-center justify-center">
      <div className="mx-auto flex flex-col px-3">
        <div className="text-center text-3xl font-thin text-gray-800 dark:text-gray-100">
          <div className="flex flex-col items-center">
            <div className="flex flex-row justify-center items-end">
              <div className="flex-shrink-0 flex flex-col items-center">
                <Image
                  src={theme === 'dark' ? darkTextLogo : lightTextLogo}
                  alt="MSF Logo"
                  priority
                  style={{
                    maxWidth: '150px',
                    maxHeight: '150px',
                  }}
                />
              </div>
            </div>
            <div className="mt-8 flex justify-center w-full">
              <SuggestedPrompts onSelectPrompt={onSelectPrompt} />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
