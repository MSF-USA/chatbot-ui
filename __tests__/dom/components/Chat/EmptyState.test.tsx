import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';

import { EmptyState } from '@/components/Chat/EmptyState/EmptyState';

// Mock next-intl
vi.mock('next-intl', () => ({
  useTranslations: () => (key: string) => key,
}));

// Mock useUI hook
vi.mock('@/lib/hooks/ui/useUI', () => ({
  useUI: () => ({
    theme: 'light',
  }),
}));

// Mock next/image
vi.mock('next/image', () => ({
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img {...props} />;
  },
}));

describe('EmptyState', () => {
  it('renders empty state', () => {
    const { container } = render(<EmptyState />);
    expect(container.firstChild).toBeInTheDocument();
  });

  it('displays MSF logo', () => {
    render(<EmptyState />);

    const logo = screen.getByAltText('MSF Logo');
    expect(logo).toBeInTheDocument();
  });

  it('renders SuggestedPrompts component', () => {
    render(<EmptyState />);

    // SuggestedPrompts renders buttons
    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBeGreaterThan(0);
  });

  it('passes onSelectPrompt to SuggestedPrompts', () => {
    const mockOnSelectPrompt = vi.fn();
    render(<EmptyState onSelectPrompt={mockOnSelectPrompt} />);

    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBeGreaterThan(0);

    // The prop should be passed down (SuggestedPrompts tests verify it works)
  });

  it('has correct layout structure', () => {
    const { container } = render(<EmptyState />);

    const wrapper = container.firstChild;
    expect(wrapper).toHaveClass('flex');
    expect(wrapper).toHaveClass('h-[88%]');
    expect(wrapper).toHaveClass('items-center');
    expect(wrapper).toHaveClass('justify-center');
  });

  it('has correct text styling', () => {
    const { container } = render(<EmptyState />);

    const textContainer = container.querySelector('.text-3xl');
    expect(textContainer).toBeInTheDocument();
    expect(textContainer).toHaveClass('font-thin');
    expect(textContainer).toHaveClass('text-gray-800');
    expect(textContainer).toHaveClass('dark:text-gray-100');
  });

  it('logo has correct max dimensions', () => {
    render(<EmptyState />);

    const logo = screen.getByAltText('MSF Logo');
    expect(logo).toHaveStyle({ maxWidth: '150px' });
    expect(logo).toHaveStyle({ maxHeight: '150px' });
  });

  it('centers logo and content', () => {
    const { container } = render(<EmptyState />);

    const centerDiv = container.querySelector('.text-center');
    expect(centerDiv).toBeInTheDocument();

    const flexCol = container.querySelector('.flex-col.items-center');
    expect(flexCol).toBeInTheDocument();
  });

  it('has margin between logo and prompts', () => {
    const { container } = render(<EmptyState />);

    const promptsWrapper = container.querySelector('.mt-8');
    expect(promptsWrapper).toBeInTheDocument();
  });

  it('works with dark theme', () => {
    // The component uses useUI hook which is already mocked at module level
    // Just verify it renders without errors
    render(<EmptyState />);

    const logo = screen.getByAltText('MSF Logo');
    expect(logo).toBeInTheDocument();
  });

  it('works without onSelectPrompt callback', () => {
    expect(() => render(<EmptyState />)).not.toThrow();
  });

  it('has proper responsive design classes', () => {
    const { container } = render(<EmptyState />);

    const mxAuto = container.querySelector('.mx-auto');
    expect(mxAuto).toBeInTheDocument();

    const px3 = container.querySelector('.px-3');
    expect(px3).toBeInTheDocument();
  });

  it('logo renders with priority prop passed to Image component', () => {
    // Note: next/image's priority prop is a Next.js optimization
    // In tests with our mock, it gets passed to the img element
    // The important thing is that the logo renders
    render(<EmptyState />);

    const logo = screen.getByAltText('MSF Logo');
    expect(logo).toBeInTheDocument();
    // The priority optimization is handled by Next.js, not visible in DOM
  });

  it('matches snapshot', () => {
    const { container } = render(<EmptyState />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
