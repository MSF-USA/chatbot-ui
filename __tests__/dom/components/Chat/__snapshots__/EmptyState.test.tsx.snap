// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyState > matches snapshot 1`] = `
<div
  class="flex h-[88%] items-center justify-center"
>
  <div
    class="mx-auto flex flex-col px-3"
  >
    <div
      class="text-center text-3xl font-thin text-gray-800 dark:text-gray-100"
    >
      <div
        class="flex flex-col items-center"
      >
        <div
          class="flex flex-row justify-center items-end"
        >
          <div
            class="flex-shrink-0 flex flex-col items-center"
          >
            <img
              alt="MSF Logo"
              src="/public/international_logo_black.png"
              style="max-width: 150px; max-height: 150px;"
            />
          </div>
        </div>
        <div
          class="mt-8 flex justify-center w-full"
        >
          <div
            class="hidden sm:flex space-x-5"
          >
            <button
              class="bg-transparent text-black dark:text-white border border-[#E0E0E0] dark:border-[#444444] rounded-md px-2 py-1 text-sm hover:bg-[#F9F9F9] dark:hover:bg-[#2F2F2F] dark:hover:text-white transition"
              style="width: 200px; height: 100px; text-align: start; white-space: normal; display: flex; flex-direction: column; align-items: start; justify-content: center; padding: 30px;"
            >
              <div
                class="flex flex-col items-start"
              >
                <svg
                  class="tabler-icon tabler-icon-language h-5 w-5 mb-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 6.371c0 4.418 -2.239 6.629 -5 6.629"
                  />
                  <path
                    d="M4 6.371h7"
                  />
                  <path
                    d="M5 9c0 2.144 2.252 3.908 6 4"
                  />
                  <path
                    d="M12 20l4 -9l4 9"
                  />
                  <path
                    d="M19.1 18h-6.2"
                  />
                  <path
                    d="M6.694 3l.793 .582"
                  />
                </svg>
                <div>
                  <span>
                    Translate Something
                  </span>
                </div>
              </div>
            </button>
            <button
              class="bg-transparent text-black dark:text-white border border-[#E0E0E0] dark:border-[#444444] rounded-md px-2 py-1 text-sm hover:bg-[#F9F9F9] dark:hover:bg-[#2F2F2F] dark:hover:text-white transition"
              style="width: 200px; height: 100px; text-align: start; white-space: normal; display: flex; flex-direction: column; align-items: start; justify-content: center; padding: 30px;"
            >
              <div
                class="flex flex-col items-start"
              >
                <svg
                  class="tabler-icon tabler-icon-mail h-5 w-5 mb-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 7a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-10z"
                  />
                  <path
                    d="M3 7l9 6l9 -6"
                  />
                </svg>
                <div>
                  <span>
                    Draft an Email
                  </span>
                </div>
              </div>
            </button>
            <button
              class="bg-transparent text-black dark:text-white border border-[#E0E0E0] dark:border-[#444444] rounded-md px-2 py-1 text-sm hover:bg-[#F9F9F9] dark:hover:bg-[#2F2F2F] dark:hover:text-white transition"
              style="width: 200px; height: 100px; text-align: start; white-space: normal; display: flex; flex-direction: column; align-items: start; justify-content: center; padding: 30px;"
            >
              <div
                class="flex flex-col items-start"
              >
                <svg
                  class="tabler-icon tabler-icon-bulb h-5 w-5 mb-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 12h1m8 -9v1m8 8h1m-15.4 -6.4l.7 .7m12.1 -.7l-.7 .7"
                  />
                  <path
                    d="M9 16a5 5 0 1 1 6 0a3.5 3.5 0 0 0 -1 3a2 2 0 0 1 -4 0a3.5 3.5 0 0 0 -1 -3"
                  />
                  <path
                    d="M9.7 17l4.6 0"
                  />
                </svg>
                <div>
                  <span>
                    Brainstorm Ideas
                  </span>
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
